### PR TITLE
[flink] Support showing physical column comments for DESC statement

### LINKED ISSUE #784: [Feature] Fix column comments for flink-1.17
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

subtask of #640 

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by yuzelin**
### Purpose

Fix #784 

### Tests

`ReadWriteTableITCase.testPhysicalColumnComments`

### API and Format 

No

### Documentation

No

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/ReadWriteTableITCase.java [hunk 1]
```diff
@@ -1331,6 +1334,22 @@ public void testStreamingInsertOverwrite() {
                 .hasMessage("Paimon doesn't support streaming INSERT OVERWRITE.");
     }
 
+    @Test
+    public void testPhysicalColumnComments() {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/ReadWriteTableITCase.java:1338]: add a case for `testComputedColumnComments` too? We would see no comments return.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/DataCatalogTable.java [hunk 2]
```diff
@@ -50,6 +54,17 @@ public Table table() {
...
 
+    @Override
+    public Schema getUnresolvedSchema() {
+        // add physical column comments
+        Map<String, String> columnComments =
+                table.rowType().getFields().stream()
+                        .filter(dataField -> dataField.description() != null)
+                        .collect(Collectors.toMap(DataField::name, DataField::description));
+
+        return FlinkTableSchemaUtils.toSchema(getSchema(), columnComments);
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/DataCatalogTable.java:65]: Put `FlinkTableSchemaUtils.toSchema` to `DataCatalogTable.toSchema`? No other use it.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java [hunk 3]
```diff
@@ -364,13 +366,33 @@ public static Schema fromCatalogTable(CatalogTable catalogTable) {
...
                 options,
                 catalogTable.getComment());
     }
 
+    private static Map<String, String> getColumnComments(CatalogTable catalogTable) {
+        return catalogTable.getUnresolvedSchema().getColumns().stream()
+                .filter(c -> c.getComment().isPresent())
+                .collect(
+                        Collectors.toMap(
+                                org.apache.flink.table.api.Schema.UnresolvedColumn::getName,
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:381]: Import `org.apache.flink.table.api.Schema.UnresolvedColumn`.

Comment (yuzelin) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:381]: good catch

**Comment by yuzelin** [comment]
17EB401A-FF20-472D-9139-DD22113EBE7F

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**